bus-mapping/src/evm/opcodes.rs

```diff
@@ -425,6 +416,25 @@ pub fn gen_begin_tx_ops(state: &mut CircuitInputStateRef) -> Result<ExecStep, Er
     } else {
         (Word::zero(), true)
     };
+    if !state.is_precompiled(&call.address) && !call.is_create() {
+        state.account_read(
+            &mut exec_step,
+            call.address,
+            AccountField::CodeHash,
+            callee_code_hash,
+        );
+    }
+
+    // Transfer with fee
+    state.transfer_with_fee(
+        &mut exec_step,
+        call.caller_address,
+        call.address,
+        callee_exists,
+        call.is_create(),
```

Comment:
i think even though is_create == true, must_create may not be true?   A user can transfer eth to a pre-calculated address before he really deploy contract at that address

Reply:
Kimi also thought about this case when implementing `CREATE`, and we discussed yesterday then decide to update the account `code_hash` to `EMPTH_HASH` no matter what, but let the prover witness the previous value, and constrain it shuold be either `0` or `EMPTH_HASH`

(or in another way we could ask prover the witness `callee_exists` to select `EMPTH_HASH` or `0` as previous value).